STYLE_GUIDE.md

```diff
@@ -1,3 +1,66 @@
 # Fortran stdlib Style Guide
 
-This document will describe the code style to use.
+> [A]utomate and codify as much as you possibly can while remembering that the human touch is still necessary to praise
+> and be kind to your contributors.
+> Let robots handle your project’s pedantry and humans handle your project’s empathy.
+
+-- @mikemcquaid, [Homebrew] project leader<sup>[1]</sup>
+
+[1]: https://mikemcquaid.com/2018/06/05/robot-pedantry-human-empathy/
+[Homebrew]: https://brew.sh
+
+## File naming conventions
+
+- Source files should contain at most one `program` or `module`/`submodule`.
+- The filename should match the program or module name and have the file extension `.f90` or `.F90` if preprocessing is required.
+- If the interface and implementation is split using submodules the implementation submodule file should have the same name as the interface (parent) module but end in `_implementation`. E.g., `string_class.f90` and `string_class_implementation.f90`
+- Tests should be added in the `tests` subdirectory and have the same name as the module they are testing with the `test_` prefix added. E.g., `string_class.f90` and `tests/test_string_class.f90`
+
+## Indentation & whitespace
+
+By setting and following a convention for indentation and whitespace, code reviews and git-diffs can
+focus on the semantics of the proposed changes rather than style and formatting.
+
+* __TABs vs spaces__: __Spaces__: TABs should never be used for indentation
+* __Trailing whitespace__: All trailing whitespace must be removed
+* __End of File (EOF)__: The file must end in a single new-line character
+* __Line length__: __132__: The Fortran 90 (and later) standard mandates lines be no longer than 132 characters
+* __Indentation__: __Four (4) spaces__ for all control constructs, line continuations, etc.
+* __Line ending character__: __Native__: Git will convert line endings to `\n` (Unix style) on commit, and will checkout files with native line endings
+
+## Variable naming
+
+Variable names should be descriptive, and not artificially short.
+By default, where it makes sense to do so, variable names shall be made up of one or more full words separated by an underscore.
+For cases where a conventional & appropriate shortening of a word is used then the underscore may be omitted.
```

Comment:
I suggest to rework this section. 

Here is a suggestion (from https://www.fortran90.org/src/best-practices.html#naming-convention):

---

* Use lowercase for all Fortran constructs (`do`, `subroutine`, `module`, ...).
* Follow short mathematical notation for mathematical variables/functions (`Ylm`, `Gamma`, `gamma`, `Enl`, `Rnl`, ...).
* For other names use all lowercase: try to keep names to one or two syllables; if more are required, use underscores to clarify (`sortpair`, `whitechar`, `meshexp`, `numstrings`, `linspace`, `meshgrid`, `argsort`, `spline`, `spline_interp`, `spline_interpolate`, `stoperr`, `stop_error`, `meshexp_der`).

For example "spline interpolation" can be shortened to `spline_interpolation`, `spline_interpolate`, `spline_interp`, `spline`, but not to `splineint` ("int" could mean integration, integer, etc. --- too much ambiguity, even in the clear context of a computational code). This is in contrast to `get_argument()` where `getarg()` is perfectly clean and clear.

The above are general guidelines. In general, choosing the right name certainly depends on the word being truncated as to whether the first syllable is sufficient. Usually it is but clearly not always. Thus some thought should go into step "try to keep names to 2 syllables or less" since it can really affect the indicativeness and simplicity. Simple consistent naming rules are a real help in this regard – for both collaboration and for one’s own sanity when going back to some old code you haven’t seen in while.

---

Here is the source code so that you can copy & paste it:
```markdown
* Use lowercase for all Fortran constructs (`do`, `subroutine`, `module`, ...).
* Follow short mathematical notation for mathematical variables/functions (`Ylm`, `Gamma`, `gamma`, `Enl`, `Rnl`, ...).
* For other names use all lowercase: try to keep names to one or two syllables; if more are required, use underscores to clarify (`sortpair`, `whitechar`, `meshexp`, `numstrings`, `linspace`, `meshgrid`, `argsort`, `spline`, `spline_interp`, `spline_interpolate`, `stoperr`, `stop_error`, `meshexp_der`).

For example "spline interpolation" can be shortened to `spline_interpolation`, `spline_interpolate`, `spline_interp`, `spline`, but not to `splineint` ("int" could mean integration, integer, etc. --- too much ambiguity, even in the clear context of a computational code). This is in contrast to `get_argument()` where `getarg()` is perfectly clean and clear.

The above are general guidelines. In general, choosing the right name certainly depends on the word being truncated as to whether the first syllable is sufficient. Usually it is but clearly not always. Thus some thought should go into step "try to keep names to 2 syllables or less" since it can really affect the indicativeness and simplicity. Simple consistent naming rules are a real help in this regard – for both collaboration and for one’s own sanity when going back to some old code you haven’t seen in while.
```

Reply:
I followed @milancurcic's suggestions from below but I'm happy to iterate on this more too. I'm marking as resolved because its outdated and changed, but if you feel strongly that what is there now needs changing, please don't hesitate to say so.